### PR TITLE
[runtime] Default Interface method fixes.

### PR DESCRIPTION
**Pull request by vargaz**
Contributes to #11267

### CODE REVIEW

**Comment by lambdageek** [comment]
Looks like interpreter needs corresponding fixes.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/mini/interp/transform.c [hunk 1]
```diff
@@ -1596,12 +1600,28 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
...
+				if (mint_type (m_class_get_byval_arg (constrained_class)) == MINT_TYPE_VT) {
+					ADD_CODE (td, MINT_BOX_VT);
+					ADD_CODE (td, get_data_item_index (td, constrained_class));
+					ADD_CODE (td, csignature->param_count | ((td->sp - 1 - csignature->param_count)->type != STACK_TYPE_MP ? 0 : BOX_NOT_CLEAR_VT_SP));
+				} else {
+					ADD_CODE (td, MINT_BOX);
+					ADD_CODE (td, get_data_item_index (td, constrained_class));
+					ADD_CODE (td, csignature->param_count);
+				}
+			}
```
Comment (lewurm) [inline mono/mini/interp/transform.c]: feels like code duplication, but then again I'm not sure about all those conditions...

Comment (vargaz) [reply mono/mini/interp/transform.c]: Yes, there is some code duplication, but the conditionals are a bit different.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**